modules/apps/blogs/blogs-web/src/main/resources/META-INF/resources/blogs/js/blogs.js

```diff
@@ -12,769 +12,616 @@
+
+			this._bindUI();
+
+			const draftEntry = entry?.status === constants.STATUS_DRAFT;
+
+			const userEntry = entry?.userId === themeDisplay.getUserId();
+
+			if (!entry || (userEntry && draftEntry)) {
+				this._initDraftSaveInterval();
+			}
+
+			const customDescriptionEnabled = entry?.customDescription;
+
+			this._customDescription = customDescriptionEnabled
+				? entry.description
+				: STR_BLANK;
+			this._shortenDescription = !customDescriptionEnabled;
+
+			if (emailEntryUpdatedEnabled) {
+				Liferay.Util.toggleBoxes(
+					`${namespace}sendEmailEntryUpdated`,
+					`${namespace}emailEntryUpdatedCommentWrapper`
+				);
+			}
+
+			window[`${namespace}onChangeContentEditor`] = (html) => {
+				this.setDescription(html);
+			};
+		});
+	}
 
-					var customAbstractOptions = instance.one(
-						'#entryAbstractOptions'
-					);
-
-					if (customAbstractOptions) {
-						eventHandles.push(
-							customAbstractOptions.delegate(
-								STR_CHANGE,
-								instance._configureAbstract,
-								'input[type="radio"]',
-								instance
-							)
-						);
-					}
+	_automaticURL() {
+		const automaticURLInput = document.querySelector(
+			`input[name=${this._config.namespace}automaticURL]:checked`
+		);
 
-					var urlOptions = instance.one('#urlOptions');
-
-					eventHandles.push(
-						urlOptions.delegate(
-							STR_CHANGE,
-							instance._onChangeURLOptions,
-							'input[type="radio"]',
-							instance
-						)
-					);
-
-					instance._eventHandles = eventHandles;
-				},
-
-				_checkImagesBeforeSave(draft, ajax) {
-					var instance = this;
-
-					if (instance._hasTempImages()) {
-						if (
-							confirm(
-								instance.get('strings').confirmDiscardImages
-							)
-						) {
-							instance._getTempImages().each((node) => {
-								node.ancestor().remove();
-							});
-
-							instance._saveEntry(draft, ajax);
-						}
-					}
-					else {
-						instance._saveEntry(draft, ajax);
-					}
-				},
+		return automaticURLInput.value === 'true';
+	}
 
-				_configureAbstract(event) {
-					var instance = this;
+	_beforePublishBtnClick() {
+		const form = Liferay.Form.get(`${this._config.namespace}fm`);
 
-					var target = event.target;
+		form.addRule(
+			`${this._config.namespace}title`,
+			'required',
+			this._config.strings.titleRequiredAtPublish
+		);
+	}
 
-					var description = instance._customDescription;
+	_beforeSaveBtnClick() {
+		const form = Liferay.Form.get(`${this._config.namespace}fm`);
 
-					instance._shortenDescription = target.val() === 'false';
+		form.removeRule(`${this._config.namespace}title`, 'required');
+	}
 
-					if (instance._shortenDescription) {
-						instance._customDescription = instance
-							.one('#description')
-							.val();
+	_bindUI() {
+		const events = this._events;
+
+		this._captionNode = this._rootNode.querySelector(
+			'.cover-image-caption'
+		);
+
+		Liferay.on('coverImageDeleted', this._removeCaption, this);
+		Liferay.on(
```

Comment:
These listeners are the ones that are truly risky since the `Liferay` object is not destroyed on navigation. Failing to detach them will cause issues when the callback will execute multiple times.